Support building with Muon buildsystem

[Muon buildsystem](https://sr.ht/~lattis/muon/) is the Meson-compatible implementation in pure C language. It supports the majority of the Meson features.

With the following changes latest Muon from git successfully finishes `muon setup build`:
```patch
diff --git a/meson.build b/meson.build
index bd968f79ad..ad35b04a26 100644
--- a/meson.build
+++ b/meson.build
@@ -11,7 +11,7 @@ project('rizin', 'c',
 py3_exe = import('python').find_installation()
 git_exe = find_program('git', required: false)
 pkgconfig_mod = import('pkgconfig')
-cmake_mod = import('cmake')
+cmake_mod = import('cmake', required: false)
 
 # Python scripts used during the build process
 create_tags_rz_py = files('sys/create_tags_rz.py')
diff --git a/subprojects/libuv.wrap b/subprojects/libuv.wrap
index e290982083..4e4198e528 100644
--- a/subprojects/libuv.wrap
+++ b/subprojects/libuv.wrap
@@ -1,9 +1,8 @@
 [wrap-file]
 directory = libuv-v1.40.0
 
-source_url = https://raw.githubusercontent.com/rizinorg/fallback-repo/main/libuv-v1.40.0.tar.gz
 source_filename = libuv-v1.40.0.tar.gz
 source_hash = 61a90db95bac00adec1cc5ddc767ebbcaabc70242bd1134a7a6b1fb1d498a194
-source_fallback_url = https://dist.libuv.org/dist/v1.40.0/libuv-v1.40.0.tar.gz
+source_url = https://dist.libuv.org/dist/v1.40.0/libuv-v1.40.0.tar.gz
 
 patch_directory = libuv-v1.40.0
diff --git a/subprojects/zlib.wrap b/subprojects/zlib.wrap
index 41923456fe..9284d5bd2c 100644
--- a/subprojects/zlib.wrap
+++ b/subprojects/zlib.wrap
@@ -1,7 +1,6 @@
 [wrap-file]
 directory = zlib-1.2.12
-source_url = https://raw.githubusercontent.com/rizinorg/fallback-repo/main/zlib-1.2.12.tar.gz
 source_filename = zlib-1.2.12.tar.gz
 source_hash = 91844808532e5ce316b3c010929493c0244f3d37593afd6de04f71821d5136d9
 patch_directory = zlib-1.2.12
-source_fallback_url = https://zlib.net/fossils/zlib-1.2.12.tar.gz
+source_url = https://zlib.net/fossils/zlib-1.2.12.tar.gz
```
However, during the `ninja -C build` phase there is still error remain:
```
cc -std=c99 -g -Og -Wpedantic -Wextra -Wall -Werror=sizeof-pointer-memaccess -I subprojects/libzip-1.7.3 -I ../subprojects/libzip-1.7.3 -I ../subprojects/libzip-1.7.3/lib -fpic -MD -MQ subprojects/libzip-1.7.3/liblibzip.a.p/lib/zip_name_locate.c.o -MF subprojects/libzip-1.7.3/liblibzip.a.p/lib/zip_name_locate.c.o.d -o subprojects/libzip-1.7.3/liblibzip.a.p/lib/zip_name_locate.c.o -c ../subprojects/libzip-1.7.3/lib/zip_name_locate.c
../subprojects/libzip-1.7.3/lib/zip_name_locate.c: In function â€˜_zip_name_locateâ€™:
../subprojects/libzip-1.7.3/lib/zip_name_locate.c:65:41: error: â€˜strcasecmpâ€™ undeclared (first use in this function); did you mean â€˜strncmpâ€™?
   65 |         cmp = (flags & ZIP_FL_NOCASE) ? strcasecmp : strcmp;
      |                                         ^~~~~~~~~~
      |                                         strncmp
../subprojects/libzip-1.7.3/lib/zip_name_locate.c:65:41: note: each undeclared identifier is reported only once for each function it appears in
[1251/1781] compiling c subprojects/libzip-1.7.3/liblibzip.a.p/lib/zip_memdup.c.o
```
Full Ninja log here: 
[muon_ninja.zip](https://github.com/rizinorg/ideas/files/8857287/muon_ninja.zip)

cc @wargio @ret2libc @thestr4ng3r

@ret2libc latest particular bug was fixed but linking still fails:
```
ninja: Entering directory `build'
[1/644] linking static subprojects/tree-sitter-0.20.6/libtree_sitter.a
[2/644] linking static subprojects/zlib-1.2.12/libz.a
[3/644] linking static subprojects/lz4-1.9.3/liblz4.a
[4/644] compiling c subprojects/xz-5.2.5/src/liblzma/liblzma.a.p/common/easy_preset.c.o
[5/644] linking subprojects/libuv-v1.40.0/libuv.so
FAILED: subprojects/libuv-v1.40.0/libuv.so 
cc  -o subprojects/libuv-v1.40.0/libuv.so subprojects/libuv-v1.40.0/libuv.so.p/src/fs-poll.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/idna.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/inet.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/random.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/strscpy.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/threadpool.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/timer.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/uv-common.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/uv-data-getter-setters.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/version.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/async.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/core.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/dl.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/fs.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/getaddrinfo.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/getnameinfo.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/loop-watcher.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/loop.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/pipe.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/poll.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/process.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/random-devurandom.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/signal.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/stream.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/tcp.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/thread.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/tty.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/udp.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/linux-core.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/linux-inotify.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/linux-syscalls.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/procfs-exepath.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/pthread-fixes.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/random-getrandom.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/random-sysctl-linux.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/proctitle.c.o -pthread -shared -Wl,-soname,libuv.so -Wl,--as-needed -Wl,--no-undefined -Wl,--start-group /usr/lib64/libm.so /usr/lib64/libdl.a -Wl,--end-group
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/fs-poll.c.o: relocation R_X86_64_32 against `.text' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/idna.c.o: relocation R_X86_64_32S against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/inet.c.o: relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/random.c.o: relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/threadpool.c.o: relocation R_X86_64_32 against `.bss' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/timer.c.o: relocation R_X86_64_32 against `.text' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/uv-common.c.o: relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/uv-data-getter-setters.c.o: relocation R_X86_64_32S against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/version.c.o: relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/async.c.o: relocation R_X86_64_32 against `.text' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/core.c.o: relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/dl.c.o: relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/fs.c.o: relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/getaddrinfo.c.o: relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/getnameinfo.c.o: relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/loop.c.o: relocation R_X86_64_32 against symbol `uv__work_done' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/pipe.c.o: relocation R_X86_64_32S against symbol `uv__server_io' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/poll.c.o: relocation R_X86_64_32 against `.text' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/process.c.o: relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/random-devurandom.c.o: relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/signal.c.o: relocation R_X86_64_32 against `.bss' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/stream.c.o: relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/tcp.c.o: relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/thread.c.o: relocation R_X86_64_32 against `.text' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/tty.c.o: relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/udp.c.o: relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/linux-core.c.o: relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/linux-inotify.c.o: relocation R_X86_64_32 against `.text' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/procfs-exepath.c.o: relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/random-sysctl-linux.c.o: relocation R_X86_64_32S against `.data' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/proctitle.c.o: relocation R_X86_64_32 against `.bss' can not be used when making a shared object; recompile with -fPIC
collect2: error: ld returned 1 exit status
[6/644] compiling c subprojects/xz-5.2.5/src/liblzma/liblzma.a.p/common/filter_common.c.o
[7/644] compiling c subprojects/xz-5.2.5/src/liblzma/liblzma.a.p/common/stream_flags_common.c.o
[8/644] compiling c subprojects/xz-5.2.5/src/liblzma/liblzma.a.p/common/vli_size.c.o
[9/644] compiling c subprojects/xz-5.2.5/src/liblzma/liblzma.a.p/common/hardware_physmem.c.o
[10/644] compiling c subprojects/xz-5.2.5/src/liblzma/liblzma.a.p/common/auto_decoder.c.o
[11/644] compiling c subprojects/xz-5.2.5/src/liblzma/liblzma.a.p/common/easy_decoder_memusage.c.o
[12/644] compiling c subprojects/xz-5.2.5/src/liblzma/liblzma.a.p/common/block_buffer_decoder.c.o
[13/644] compiling c subprojects/xz-5.2.5/src/liblzma/liblzma.a.p/common/block_header_decoder.c.o
[14/644] linking static librz/core/librz_core.a
[15/644] compiling c subprojects/xz-5.2.5/src/liblzma/liblzma.a.p/common/filter_buffer_decoder.c.o
[16/644] compiling c subprojects/xz-5.2.5/src/liblzma/liblzma.a.p/common/index.c.o
[17/644] compiling c subprojects/xz-5.2.5/src/liblzma/liblzma.a.p/common/alone_decoder.c.o
[18/644] compiling c subprojects/xz-5.2.5/src/liblzma/liblzma.a.p/common/block_decoder.c.o
ninja: build stopped: subcommand failed.
```

--- a/.github/workflows/muon.yml
+++ b/.github/workflows/muon.yml
@@ -1,0 +1,75 @@
+name: Muon build
+
+on:
+  push:
+    paths:
+    - '**.c'
+    - '**.h'
+    - '**.in'
+    - '**.inc'
+    - '**/meson.build'
+    - 'subprojects/**'
+    - 'test/**'
+    - '.github/workflows/muon.yml'
+    branches:
+    - 'dev'
+    - 'stable'
+    - 'container-*'
+  pull_request:
+    paths:
+    - '**.c'
+    - '**.h'
+    - '**.in'
+    - '**.inc'
+    - '**/meson.build'
+    - 'subprojects/**'
+    - 'test/**'
+    - '.github/workflows/muon.yml'
+    branches:
+    - 'dev'
+    - 'stable'
+    - 'release-*'
+
+jobs:
+  build:
+    name: ubuntu-muon
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Disable CMake module (unsupported by Muon)
+      run: |
+        sed -i "s/import('cmake')/import('cmake', required: false)/" meson.build
+    - name: Install dependencies
+      run: |
+          sudo apt-get --assume-yes update
+          sudo apt-get --assume-yes install python3-wheel python3-setuptools pkgconf libcurl4-openssl-dev libpkgconf-dev libarchive-dev
+          sudo python3 -m pip install ninja
+
+    - name: Checkout Muon repository
+      run: |
+          git clone https://git.sr.ht/~lattis/muon
+
+    - name: Compiling and installing Muon
+      working-directory: muon
+      run: |
+          ./bootstrap.sh build
+          build/muon setup build
+          ninja -C build
+          sudo cp build/muon /usr/bin/muon
+
+    - name: Muon setup
+      run: muon setup build
+    - name: Ninja compile and install
+      run: ninja -C build && sudo muon -C build install
+
+    - name: Checkout our Testsuite Binaries
+      uses: actions/checkout@v2
+      with:
+          repository: rizinorg/rizin-testbins
+          path: test/bins
+
+    - name: Run unit tests
+      run: muon -C build test
+    - name: Rizin version
+      run: |
+        rizin -v

--- a/meson.build
+++ b/meson.build
@@ -5,6 +5,7 @@ project('rizin', 'c',
   default_options: [
     'buildtype=debugoptimized',
     'b_vscrt=from_buildtype',
+    'warning_level=1',
   ]
 )
 

